lib/CommandCall.js

```diff
@@ -53,32 +44,32 @@ class CommandCall {
    * @returns {string}
    */
   toXML() {
-    if (this.type === 'sh') {
-      if (this.options) {
-        this.xml = iXmlNodeShOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error)
-              + this.command + iXmlNodeShClose();
-      } else {
-        this.xml = iXmlNodeShOpen('', '', '', '', '') + this.command + iXmlNodeShClose();
-      }
-    } else if (this.type === 'cl') {
-      const exec = (this.command.indexOf('?') > 0) ? 'rexx' : 'cmd';
+    // consider renaming type: 'cl' -> cmd to avoid this check.
+    if (this.type === 'cl') {
+      this.type = 'cmd';
+    }
 
-      if (this.options) {
-        this.xml = iXmlNodeCmdOpen(this.options.exec, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeCmdClose();
-      } else {
-        this.xml = iXmlNodeCmdOpen(exec, '', '', '', '', '') + this.command + iXmlNodeCmdClose();
-      }
-    } else if (this.type === 'qsh') {
-      if (this.options) {
-        this.xml = iXmlNodeQshOpen(this.options.rows, this.options.hex, this.options.before,
-          this.options.after, this.options.error) + this.command + iXmlNodeQshClose();
-      } else {
-        this.xml = iXmlNodeQshOpen('', '', '', '', '') + this.command + iXmlNodeQshClose();
-      }
+    this.xml = `<${this.type}`;
+
+    if (this.type === 'cmd' && !this.options.exec) { // set default value
+      // eslint-disable-next-line no-unused-expressions
+      this.command.indexOf('?') > 0 ? this.xml += " exec='rexx'" : this.xml += " exec='cmd'";
     }
 
+    if (!this.options.error) { // set default value
+      this.xml += " error='fast'";
+    }
```

Comment:
Similar to the other review, I would like to see these a little more DRY.

Reply:
I've updated the PR